sequencer/src/l1_client.rs

```diff
@@ -276,140 +163,70 @@ async fn get_finalized_block<P: JsonRpcClient>(
 
 #[cfg(test)]
 mod test {
+
+    use crate::NodeState;
+
     use super::*;
     use async_compatibility_layer::logging::{setup_backtrace, setup_logging};
-    use futures::channel::oneshot;
+    use ethers::utils::Anvil;
     use sequencer_utils::AnvilOptions;
-    use std::time::Instant;
 
     #[async_std::test]
-    async fn test_l1_block_scanning() {
+    async fn test_l1_block_fetching() {
         setup_logging();
         setup_backtrace();
 
-        let test_duration = Duration::from_secs(30);
-        let test_interval = Duration::from_millis(500);
         let anvil = AnvilOptions::default()
             .block_time(Duration::from_secs(1))
             .spawn()
             .await;
-        let l1_client = L1ClientOptions::with_url(anvil.ws_url())
-            .start()
-            .await
-            .unwrap();
-        let anvil_client = Provider::try_from(anvil.url().to_string()).unwrap();
-
-        let mut prev_head = 0;
-        let mut prev_finalized: Option<L1BlockInfo> = None;
 
-        let start = Instant::now();
-        while start.elapsed() < test_duration {
-            let snapshot = l1_client.snapshot().await;
+        let l1_client = L1Client::new(anvil.url());
```

Comment:
We don't actually use `l1_client` in the rest of the test, so this seems to be testing the ethers-rs provider rather than testing our `L1Client`

Reply:
ef28a828d766970c14c8d2b42b1b2afbadd4347e does a better job of testing `L1Client`.